src/pages/resultsView/download/CaseAlterationTable.tsx

```diff
@@ -33,58 +39,78 @@ export interface ICaseAlterationTableProps {
-            getValues: subAlterationMapper
-        },
-        {
-            label: "EXP",
-            getAlterationData: (oqlData: IOqlData) => oqlData.mrnaExp,
-            getValues: subAlterationMapper
-        },
-        {
-            label: "PROT",
-            getAlterationData: (oqlData: IOqlData) => oqlData.proteinLevel,
-            getValues: subAlterationMapper
-        }
-    ];
+        alterationData.map((alteration: ISubAlteration) => alteration.type);        
+    let generator;
+    let oqlValue: string = "";
 
-    // for each alteration type, transform alteration data into a comma separated string
-    generators.forEach((generator) => {
-        const alterationData = generator.getAlterationData(data);
+    if (data.alterationTypes.length === 0 || !data.alterationTypes.includes(alterationType)){
+        oqlValue = "no alteration";
+    }
+    switch (alterationType){
+        case "MUT":
+            generator = {
+                label: "MUT",
+                getAlterationData: (oqlData: IOqlData) => oqlData.mutation,
+                getProfiledData: (oqlData: IOqlData) => oqlData.isMutationProfiled,
+                getValues: stringMapper
+            }
+            break;
+        case "FUSION":
+            generator = {
+                label: "FUSION",
+                getAlterationData: (oqlData: IOqlData) => oqlData.fusion,
+                getProfiledData: (oqlData: IOqlData) => oqlData.isFusionProfiled,
+                getValues: stringMapper
+            }
+            break;
+        case "CNA":
+            generator = {
+                label: "CNA",
+                getAlterationData: (oqlData: IOqlData) => oqlData.cna,
+                getProfiledData: (oqlData: IOqlData) => oqlData.isCnaProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        case "EXP":
+            generator = {
+                label: "EXP",
+                getAlterationData: (oqlData: IOqlData) => oqlData.mrnaExp,
+                getProfiledData: (oqlData: IOqlData) => oqlData.isMrnaExpProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        case "PROT":
+            generator = {
+                label: "PROT",
+                getAlterationData: (oqlData: IOqlData) => oqlData.proteinLevel,
+                getProfiledData: (oqlData: IOqlData) => oqlData.isProteinLevelProfiled,
+                getValues: subAlterationMapper
+            }
+            break;
+        default:
+    }
 
-        if (alterationData.length > 0) {
-            oqlValue.push(`${generator.label}: ${generator.getValues(alterationData).join(",")};`);
+    if (generator){
+        const alterationData = generator.getAlterationData(data);
+        if (alterationData.length > 0){
+            oqlValue = generator.getValues(alterationData).join(",");
         }
-    });
+        const ProfiledData = generator.getProfiledData(data);
```

Comment:
never start a variable name with an uppercase character. Actually I don't think you even need a variable here. Just do this:

```suggestion
        if (generator.isProfiled(data)) {
           ...
        }
```